dive_core/CMakeLists.txt

```diff
@@ -116,18 +116,18 @@ if(MSVC)
         freedreno_devices_header
         PROPERTIES FOLDER "freedreno"
     )
-endif()
+endif() # MSVC
 
 file(GLOB_RECURSE HDR_FILES "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 file(GLOB_RECURSE SRC_FILES "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp")
 list(FILTER SRC_FILES EXCLUDE REGEX "_tests.cpp$")
 list(FILTER SRC_FILES EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/tests/.*")
 
-if(DIVE_NO_DISASSEMBLY)
+if(DIVE_NO_DISASSEMBLY OR APPLE)
```

Comment:
curious, the shader lib doesn't compile for apple?

Reply:
no, it doesn't not, and I tried to fix it, but it takes way longer than I expected. might be next steps